cpp/include/cudf/detail/join/filtered_join.cuh

```diff
@@ -0,0 +1,359 @@
+
+  // Hasher for primitive row types
+  using primitive_row_hasher =
+    cudf::row::primitive::row_hasher<cudf::hashing::detail::default_hash>;
+  // Linear probing scheme with bucket size 1 for primitive types
+  using primitive_probing_scheme = cuco::linear_probing<1, hasher_adapter>;
+  // Equality comparator for primitive rows
+  using primitive_row_comparator = cudf::row::primitive::row_equality_comparator;
+
+  // Hasher for complex row types with dynamic null handling
+  using row_hasher =
+    cudf::experimental::row::hash::device_row_hasher<cudf::hashing::detail::default_hash,
+                                                     nullate::DYNAMIC>;
+  // Linear probing scheme with bucket size 4 for nested data structures
+  using nested_probing_scheme = cuco::linear_probing<4, hasher_adapter>;
+  // Linear probing scheme with bucket size 1 for simple data
+  using simple_probing_scheme = cuco::linear_probing<1, hasher_adapter>;
+  // Equality comparator for complex rows with null handling and NaN comparison
+  using row_comparator = cudf::experimental::row::equality::device_row_comparator<
+    true,
+    cudf::nullate::DYNAMIC,
+    cudf::experimental::row::equality::nan_equal_physical_equality_comparator>;
+
+  storage_type _bucket_storage;  ///< Storage for hash table buckets
+
+  // Empty sentinel key used to mark empty slots in the hash table
+  static constexpr auto empty_sentinel_key = cuco::empty_key{
+    cuco::pair{std::numeric_limits<hash_value_type>::max(), lhs_index_type{JoinNoneValue}}};
+  build_properties _build_props;           ///< Properties of the build table
+  cudf::table_view _build;                 ///< input table to build the hash map
+  cudf::null_equality const _nulls_equal;  ///< whether to consider nulls as equal
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table>
+    _preprocessed_build;  ///< input table preprocssed for row operators
+
+ public:
+  /**
+   * @brief Calculates the required storage size for the hash table
+   *
+   * Computes the appropriate size for the bucket storage based on the input
+   * table size and desired load factor.
+   *
+   * @param tbl Table for which to calculate storage
+   * @param load_factor Target load factor for the hash table
+   * @return Calculated bucket storage size
+   */
+  auto compute_bucket_storage_size(cudf::table_view tbl, double load_factor);
+
+ public:
+  /**
+   * @brief Constructor for filtered_join base class
+   *
+   * Initializes the hash table with the build table and prepares it for join operations.
+   *
+   * @param build The table to build the hash table from
+   * @param compare_nulls How null values should be compared
+   * @param load_factor Target load factor for the hash table
+   * @param stream CUDA stream on which to perform operations
+   */
+  filtered_join(cudf::table_view const& build,
+                cudf::null_equality compare_nulls,
+                double load_factor,
```

Comment:
@PointKernel do we need this `load_factor` exposed? Can we just set a default value?